### PR TITLE
Refine the document of BCELoss and add functional binary_cross_entropy.

### PR DESCRIPTION
**Pull request by ZHUI**
修复 class api bceloss 的中文文档。

### CODE REVIEW

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 1]
```diff
@@ -33,47 +33,55 @@ BCELoss
...
-  - **reduction(str, optional)**：- 指定应用于输出结果的计算方式，可选值有: ``'none'``, ``'mean'``, ``'sum'`` 。默认为 ``'mean'``，计算 `BCELoss` 的均值；设置为 ``'sum'`` 时，计算 `BCELoss` 的总和；设置为 ``'none'`` 时，则返回BCELoss。
+参数
+:::::::::
+  - **weight(Tensor, optional)** - 手动指定每个batch二值交叉熵的权重，如果指定的话，维度必须是一个batch的数据的维度。数据类型是float32, float64。默认是：None。
+  - **reduction(str, optional)** - 指定应用于输出结果的计算方式，可选值有: ``'none'``, ``'mean'``, ``'sum'`` 。默认为 ``'mean'``，计算 `BCELoss` 的均值；设置为 ``'sum'`` 时，计算 `BCELoss` 的总和；设置为 ``'none'`` 时，则返回BCELoss。
 
-返回：返回计算BCELoss的可调用对象。
+形状
+:::::::::
+    - **x** - :math:`(N, *)` ，输入 ``x`` 的维度是[N, *], 其中N是batch_size， `*` 是任意其他维度。
```
Comment (ZHUI) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: (Tensor), dtype, 输入数据 ``x`` 一般是 ``sigmoid`` 的输出

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 2]
```diff
@@ -33,47 +33,55 @@ BCELoss
...
 
-**注意：输入数据一般是 `fluid.layers.sigmoid` 的输出。因为是二分类，所以标签值应该是0或者1。
+**注意**：输入数据 ``x`` 一般是 ``sigmoid`` 的输出。因为是二分类，所以标签值 ``label`` 应该是0或者1。
 
-输入input和标签label的维度是[N, *], 其中N是batch_size， `*` 是任意其他维度。
-如果 :attr:`reduction` 是 ``'none'``, 则输出的维度为 [N, *], 与输入input的形状相同。
-如果 :attr:`reduction` 是 ``'mean'`` 或 ``'sum'``, 则输出的维度为 [1]。
 
-参数：
-  - **weight(Variable, optional)**：- 手动指定每个batch二值交叉熵的权重，如果指定的话，维度必须是一个batch的数据的维度。数据类型是float32, float64。默认是：None。
```
Comment (ZHUI) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst:43]: optional -> 可选

**Comment by ZHUI** [comment]
image
image
image

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 3]
```diff
@@ -33,47 +33,54 @@ BCELoss
...
+  - **reduction** (str, 可选) - 指定应用于输出结果的计算方式，可选值有: ``'none'``, ``'mean'``, ``'sum'`` 。默认为 ``'mean'``，计算 `BCELoss` 的均值；设置为 ``'sum'`` 时，计算 `BCELoss` 的总和；设置为 ``'none'`` 时，则返回BCELoss。
 
-参数：
-  - **weight(Variable, optional)**：- 手动指定每个batch二值交叉熵的权重，如果指定的话，维度必须是一个batch的数据的维度。数据类型是float32, float64。默认是：None。
-  - **reduction(str, optional)**：- 指定应用于输出结果的计算方式，可选值有: ``'none'``, ``'mean'``, ``'sum'`` 。默认为 ``'mean'``，计算 `BCELoss` 的均值；设置为 ``'sum'`` 时，计算 `BCELoss` 的总和；设置为 ``'none'`` 时，则返回BCELoss。
+形状
+:::::::::
+    - **x** (Tensor) - :math:`(N, *)` , 其中N是batch_size， `*` 是任意其他维度。输入数据 ``x`` 一般是 ``sigmoid`` 的输出。数据类型是float32、float64。
+    - **label** (Tensor) - :math:`(N, *)` ，标签 ``label`` 的维度、数据类型与输入 ``x`` 相同。
+    - **out** (Tensor) - 如果 :attr:`reduction` 是 ``'none'``, 则输出的维度为 :math:`(N, *)` , 与输入 ``x`` 的形状相同。如果 :attr:`reduction` 是 ``'mean'`` 或 ``'sum'``, 则输出的维度为 :math:`(1)` 。
```
Comment (LutaoChu) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: out看要不要用output

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 4]
```diff
@@ -33,47 +33,54 @@ BCELoss
...
     output_data = exe.run(fluid.default_main_program(),
-            feed={"input":input_data, "label":label_data},
+            feed={"x":x_data, "label":label_data},
             fetch_list=[output],
             return_numpy=True)
-    
     print(output_data)  # [array([0.65537095], dtype=float32)]
-    
+
     # imperative mode
```
Comment (LutaoChu) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: 建议把imperative示例提前

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 5]
```diff
@@ -3,26 +3,26 @@
...
 当 `weight` 为空时，公式为：
 
 .. math::
-  Out = -1 * (label * log(input) + (1 - label) * log(1 - input))
+  Out = -1 * (label * log(x) + (1 - label) * log(1 - x))
 
 当 `reduction` 为 `none` 时，最终的输出结果为：
 
 .. math::
   Out = Out
```
Comment (LutaoChu) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: 这种情况建议不写公式了吧，写到文字里就可以了

Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 4 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 3 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 6]
```diff
@@ -3,26 +3,23 @@
 BCELoss
 -------------------------------
 
-.. py:function:: paddle.nn.BCELoss(input, label, weight=None, reduction='mean')
+.. py:class:: paddle.nn.BCELoss(weight=None, reduction='mean', name=None)
 
-该接口用于创建一个BCELoss的可调用类，用于计算输入和标签之间的二值交叉熵损失值。二值交叉熵损失函数公式如下：
+该接口用于创建一个BCELoss的可调用类，用于计算输入 ``x`` 和标签 ``label`` 之间的二值交叉熵损失值。二值交叉熵损失函数公式如下：
```
Comment (wawltor) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: x-> input 同下

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 7]
```diff
@@ -33,47 +30,57 @@ BCELoss
...
+**注意**：输入数据 ``x`` 一般是 ``sigmoid`` 的输出。因为是二分类，所以标签值 ``label`` 应该是0或者1。
+
 
-输入input和标签label的维度是[N, *], 其中N是batch_size， `*` 是任意其他维度。
-如果 :attr:`reduction` 是 ``'none'``, 则输出的维度为 [N, *], 与输入input的形状相同。
-如果 :attr:`reduction` 是 ``'mean'`` 或 ``'sum'``, 则输出的维度为 [1]。
+参数
+:::::::::
+  - **weight** (Tensor，可选) - 手动指定每个batch二值交叉熵的权重，如果指定的话，维度必须是一个batch的数据的维度。数据类型是float32, float64。默认值是：None。
+  - **reduction** (str，可选) - 指定应用于输出结果的计算方式，可选值有: ``'none'``, ``'mean'``, ``'sum'`` 。默认为 ``'mean'``，计算 `BCELoss` 的均值；设置为 ``'sum'`` 时，计算 `BCELoss` 的总和；设置为 ``'none'`` 时，则返回BCELoss。
```
Comment (wawltor) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: BCELoss是一个class，binary_cross_entropy loss 或者小写的bce_loss

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 8]
```diff
@@ -33,47 +30,57 @@ BCELoss
...
+  - **name** (str，可选) - 操作的名称（可选，默认值为None）。更多信息请参见 :ref:`api_guide_Name` 。
 
-参数：
-  - **weight(Variable, optional)**：- 手动指定每个batch二值交叉熵的权重，如果指定的话，维度必须是一个batch的数据的维度。数据类型是float32, float64。默认是：None。
-  - **reduction(str, optional)**：- 指定应用于输出结果的计算方式，可选值有: ``'none'``, ``'mean'``, ``'sum'`` 。默认为 ``'mean'``，计算 `BCELoss` 的均值；设置为 ``'sum'`` 时，计算 `BCELoss` 的总和；设置为 ``'none'`` 时，则返回BCELoss。
+形状
+:::::::::
+    - **x** (Tensor) - :math:`(N, *)` , 其中N是batch_size， `*` 是任意其他维度。输入数据 ``x`` 一般是 ``sigmoid`` 的输出。数据类型是float32、float64。
+    - **label** (Tensor) - :math:`(N, *)` ，标签 ``label`` 的维度、数据类型与输入 ``x`` 相同。
+    - **output** (Tensor) - 输出的Tensor。如果 :attr:`reduction` 是 ``'none'``, 则输出的维度为 :math:`(N, *)` , 与输入 ``x`` 的形状相同。如果 :attr:`reduction` 是 ``'mean'`` 或 ``'sum'``, 则输出的维度为 :math:`(1)` 。
```
Comment (wawltor) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: :math:`(1)`  -> :math:`[1]`

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 7 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 8 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 6 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/functional_cn/binary_cross_entropy_cn.rst [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    import paddle
+    import paddle.nn.functional as F
+
+    paddle.disable_static()
+    input = paddle.to_tensor([0.5, 0.6, 0.7], dtype='float32')
+    label = paddle.to_tensor([1.0, 0.0, 1.0], dtype='float32')
+    output = F.binary_cross_entropy(input, label)
+    print(output.numpy())  # [0.65537095]
+    paddle.enable_static()
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/functional_cn/binary_cross_entropy_cn.rst]: 同理哈 这一行多余了~

File: doc/fluid/api_cn/nn_cn/functional_cn/binary_cross_entropy_cn.rst [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+:::::::::
+    - **input** (Tensor) - :math:`[N, *]` , 其中N是batch_size， `*` 是任意其他维度。输入数据 ``input`` 一般是 ``sigmoid`` 的输出。数据类型是float32、float64。
+    - **label** (Tensor) - :math:`[N, *]` ，标签 ``label`` 的维度、数据类型与输入 ``input`` 相同。
+    - **weight** (Tensor，可选) - 手动指定每个batch二值交叉熵的权重，如果指定的话，维度必须是一个batch的数据的维度。数据类型是float32, float64。默认值是：None。
+    - **reduction** (str，可选) - 指定应用于输出结果的计算方式，可选值有: ``'none'``, ``'mean'``, ``'sum'`` 。默认为 ``'mean'``，计算 `BCELoss` 的均值；设置为 ``'sum'`` 时，计算 `BCELoss` 的总和；设置为 ``'none'`` 时，则返回bce_loss。
+    - **name** (str，可选) - 操作的名称（可选，默认值为None）。更多信息请参见 :ref:`api_guide_Name` 。
+
+返回
+:::::::::
+    - **output** (Tensor) - 输出的Tensor。如果 :attr:`reduction` 是 ``'none'``, 则输出的维度为 :math:`(N, *)` , 与输入 ``input`` 的形状相同。如果 :attr:`reduction` 是 ``'mean'`` 或 ``'sum'``, 则输出的维度为 :math:`[1]` 。
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/functional_cn/binary_cross_entropy_cn.rst]: 这里应该不用写output，直接
【输出的Tensor。如果 :attr:`reduction` 是 ``'none'``, 则输出的维度为 :math:`(N, *)` , 与输入 ``input`` 的形状相同。如果 :attr:`reduction` 是 ``'mean'`` 或 ``'sum'``, 则输出的维度为 :math:`[1]` 。】

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 11]
```diff
@@ -33,47 +30,56 @@ BCELoss
   Out = SUM(Out)
 
 
-**注意：输入数据一般是 `fluid.layers.sigmoid` 的输出。因为是二分类，所以标签值应该是0或者1。
+**注意**：输入数据 ``input`` 一般是 ``sigmoid`` 的输出。因为是二分类，所以标签值 ``label`` 应该是0或者1。
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: 注意可以用.. note::的格式哈

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 12]
```diff
@@ -33,47 +30,56 @@ BCELoss
...
+    label_data = np.array([1.0, 0.0, 1.0]).astype("float32")
+
+    paddle.disable_static()
+    input = paddle.to_variable(input_data)
+    label = paddle.to_variable(label_data)
+    bce_loss = paddle.nn.loss.BCELoss()
+    output = bce_loss(input, label)
+    print(output.numpy())  # [0.65537095]
+    paddle.enable_static()
+
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: 只需要动态图示例就可以 静态图的可以删除哈

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst [hunk 13]
```diff
@@ -33,47 +30,56 @@ BCELoss
...
+    返回计算BCELoss的可调用对象。
 
-**代码示例**
+代码示例
+::::::::::
 
 .. code-block:: python
 
-    # declarative mode
     import paddle.fluid as fluid
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: 删除静态图示例后 这行也可以删除

File: doc/fluid/api_cn/nn_cn/functional_cn/binary_cross_entropy_cn.rst (hunk 10 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/functional_cn/binary_cross_entropy_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 11 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 12 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

File: doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst (hunk 13 above)
Comment (ZHUI) [reply doc/fluid/api_cn/nn_cn/loss_cn/BCELoss_cn.rst]: done

**Review by TCChenlong — APPROVED** [review]
LGTM
